src/prefect/engine.py

```diff
@@ -2433,7 +2434,12 @@ async def create_autonomous_task_run(task: Task, parameters: Dict[str, Any]) ->
         exit(1)
 
     try:
-        enter_flow_run_engine_from_subprocess(flow_run_id)
+        if PREFECT_EXPERIMENTAL_ENABLE_NEW_ENGINE.value():
+            from prefect.new_flow_engine import run_flow
+
+            asyncio.run(run_flow(flow_run_id=flow_run_id))
```

Comment:
The sync functionality introduced in https://github.com/PrefectHQ/prefect/pull/12889 makes this a little tricky since we need to call `run_flow` or `run_flow_sync` depending on if the flow is async or sync, but we don't know until we load the flow. Not sure the best way to handle that, but it might make sense to move flow loading responsibility out of the engine and have another object load the flow and invoke the correct run method.

Reply:
That's an excellent call. I think the cleanest way to do this is to move the loading of the `flow` and `flow_run` out of the execution engine's methods and into their own, separate step.  Then the engine itself maintains its current signature (you provide it the already-loaded flow and flow run) and we run those flows either sync or async as appropriate. In this PR the loading steps really don't seem to depend intrinsically on any other properties of the engine object, so I think they could safely be done ahead of time.


```python
if PREFECT_EXPERIMENTAL_ENABLE_NEW_ENGINE:
    from prefect.new_flow_engine import run_flow, run_flow_sync
    from prefect.deployments import load_flow_from_flow_run
    from prefect.flows import Flow, load_flow_from_entrypoint

    flow_run = run_sync(client.read_flow_run(flow_run_id))
    # call either load_flow_from_flow_run or load_flow_from_entrypoint, as appropriate
    flow = < get flow >
    if flow.isasync:
        run_sync(run_flow(flow, flow_run_id=flow_run_id))
    else:
        run_flow_sync(flow, flow_run_id=flow_run_id)